include/nbl/video/IGPUCommandPool.h

```diff
@@ -133,8 +111,19 @@ class NBL_API IGPUCommandPool : public core::IReferenceCounted, public IBackendO
             wipeNextCommandSize();
         }
 
+        ~CCommandSegment()
+        {
+            auto* cmd = reinterpret_cast<ICommand*>(m_data);
+            while ((cmd != segment_end()) || (cmd->getSize() != 0))
```

Comment:
please replace with `for (ICommand* cmd=begin(); cmd!=end();)`

Reply:
Resolved.